neon_data_models/models/api/llm.py

```diff
@@ -0,0 +1,194 @@
+    model: str = Field(description="Model to request")
+    max_tokens: int = Field(
+        default=512, ge=64, le=2048,
+        description="Maximum number of tokens to include in the response")
+    temperature: float = Field(
+        default=0.0, ge=0.0, le=1.0,
+        description="Temperature of response. 0 guarantees reproducibility, "
+                    "higher values increase variability. Must be `0.0` if "
+                    "`beam_search` is True")
+    repetition_penalty: float = Field(
+        default=1.0, ge=1.0, le=2.0,
+        description="Repetition penalty. Higher values limit repeated "
+                    "information in responses")
+    stream: bool = Field(
+        default=None, description="Enable streaming responses. "
+                                  "Mutually exclusive with `beam_search`.")
+    best_of: int = Field(
+        default=1,
+        description="Number of beams to use if `beam_search` is enabled.")
+    beam_search: bool = Field(
+        default=None, description="Enable beam search. "
+                                  "Mutually exclusive with `stream`.")
+    max_history: int = Field(
+        default=2, description="Maximum number of user/assistant "
+                               "message pairs to include in history context.")
+
+    @model_validator(mode='before')
+    @classmethod
+    def validate_inputs(cls, values):
+        # Neon modules previously defined `user` and `llm` keys, but Open AI
+        # specifies `assistant` in place of `llm` and is the de-facto standard
+        for idx, itm in enumerate(values.get('history', [])):
+            if itm[0] == "assistant":
+                values['history'][idx] = ("llm", itm[1])
+        return values
+
+    @model_validator(mode='after')
+    def validate_request(self):
+        # If beams are specified, make sure valid `stream` and `beam_search`
+        # values are specified
+        if self.best_of > 1:
+            if self.stream is True:
+                raise ValueError("Cannot stream with a `best_of` value "
+                                 "greater than 1")
+            if self.beam_search is False:
+                raise ValueError("Cannot have a `best_of` value other than 1 "
+                                 "if `beam_search` is False")
+            self.stream = False
+            self.beam_search = True
+        # If streaming, beam_search must be False
+        if self.stream is True:
+            if self.beam_search is True:
+                raise ValueError("Cannot enable both `stream` and "
+                                 "`beam_search`")
+            self.beam_search = False
+        # If beam search is enabled, streaming must be False
+        if self.beam_search is True:
+            if self.stream is True:
+                raise ValueError("Cannot enable both `stream` and "
+                                 "`beam_search`")
+            self.stream = False
```

Comment:
`if self.beam_search is True` not only `self.stream = False`
But also `best_of>1` or else it's not beam search